src/main.cpp

```diff
@@ -11,139 +11,217 @@
-            object.render();
-        }
-
-        glfwSwapBuffers(window);
-        glfwPollEvents();
-    }
-
-    glfwTerminate();
-    return 0;
+void render_satellites_of_satellite(Planet planet, Shader shader, glm::mat4 proj, glm::mat4 view, glm::mat4 model) {
+	for (int i = 0; i < planet.satellites.size(); i++) {
+		Planet satellite = planet.satellites.at(i);
+		model = glm::rotate(model, 8 * (float)glfwGetTime() / satellite.distance, glm::vec3(0.0f, 1.0f, 0.0f));
+		model = glm::translate(model, glm::vec3(satellite.distance, 0.0f, satellite.distance));
+		model = glm::scale(model, satellite.scale);
+		shader.set_mvp(proj * view * model);
+		satellite.render();
+	}
 }
 
+void render_satellites(Planet planet, Shader shader, glm::mat4 proj, glm::mat4 view) {
+	for (int i = 0; i < planet.satellites.size(); i++) {
+		Planet satellite = planet.satellites.at(i);
+		glm::mat4 model = glm::mat4{ 1.0f };
+		model = glm::translate(model, planet.position);
+		model = glm::rotate(model, 8 * (float)glfwGetTime() / satellite.distance, glm::vec3(0.0f, 1.0f, 0.0f));
+		model = glm::translate(model, glm::vec3(satellite.distance, 0.0f, satellite.distance));
+		model = glm::scale(model, satellite.scale);
+		shader.set_mvp(proj * view * model);
+		satellite.render();
+		if (satellite.satellites.size() != 0)render_satellites_of_satellite(satellite, shader, proj, view, model);
+	}
+}
+
+int main() {
+	if (!glfwInit()) {
+		std::cerr << "failed to initialize GLFW" << std::endl;
+		return 1;
+	}
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
+	glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+
+	GLFWwindow* window = glfwCreateWindow(WIDTH, HEIGHT, "Solar System", nullptr, nullptr);
+	if (!window) {
+		std::cerr << "failed to create GLFW window" << std::endl;
+		glfwTerminate();
+		return 1;
+	}
+
+	glfwMakeContextCurrent(window);
+
+	if (glewInit() != GLEW_OK) {
+		std::cerr << "failed to initialize GLEW" << std::endl;
+		return 1;
+	}
+
+	// aci setam ce callbackuri se apeleaza cand se misca mouseu, scrollu sau se apasa pe taste
+	glfwSetCursorPosCallback(window, mouse_callback);
+	glfwSetScrollCallback(window, scroll_callback);
+	glfwSetKeyCallback(window, key_callback);
+	// asta face cursoru sa dispara si sa ramana in fereastra. dai exit sa iesi
+	glfwSetInputMode(window, GLFW_CURSOR, GLFW_CURSOR_DISABLED);
+
+	State  state{};
+	Camera camera{ { 0.0f, 0.0f, 0.0f } };
+	// ii dam la window app-stateu, ca sa poata fi accesat in callbackuri
+	glfwSetWindowUserPointer(window, &state);
+
+	// vector de planete. in viitor cel mai probabil aici o sa initializam doar gen soarele sau ce o fi, la care ii adaugam dupa planetele
+	std::vector<Planet> planets;
+	float baseSize = 0.2f;
+	float baseDistance = 6.0f;
+	Planet sun = Planet(glm::vec3{ 0.0f, 0.0f, 0.0f }, glm::vec3{ 8.0f });
+	sun.texture = loadTexture("res/sun.jpg");
+	sun.distance = 0.0f * baseDistance;
```

Comment:
poti sa schimbi contructoru de la planeta sa fie gen
                   position               scale                  distance   texture_path
Planet sun{ {0.0f, 0.0f, 0.0f}, { 8.0f, 8.0f, 8.0f }, 0.0f       ,         "res/sun.jpg"};
ca sa nu mai faci assignurile alea doua separat
daca vrei poti sa schimbi si scaleu din vec3 doar in float pe care il aplicam pe toate axele, ca nu e ca si cum o sa avem planete ovale